nbgitpuller/static/js/index.js

```diff
@@ -42,6 +42,9 @@ GitSync.prototype.start = function() {
     if (typeof this.branch !== 'undefined' && this.branch != undefined) {
         syncUrlParams['branch'] = this.branch;
     }
+    if (typeof this.contentProvider !== 'undefined' && this.contentProvider != undefined) {
+        syncUrlParams['content_provider'] = this.contentProvider;
```

Comment:
I asked this on jupyterhub/random on gitter:

> I'm reviewing Sean Morris nbgitpuller PR, where a new query parameter is introduced. I wonder regarding the choice of case of the query line parameter.
>
> We have query parameters that are `baseUrl` and `targetpath` already. Sean has suggested `content_provider`. The other query parameter flags are just one word, making them the same for camelCalse, snake_case, and kebab-case.
>
> Do you have ideas on what case to use for consistency with various practices globally and/or practices in the JupyterHub org?

Min responded:

> I don't think it's very important, but good to be internally consistent for a given repo, at least. I tend to use snake_case, just because that's what my Python brain defaults to, but because I've planned carefully. That's what the few multi-word query params in JupyterHub use. If a repo's already adopted a camelCase convention, I'd stick with it.

Do you have energy to go for a conversion to contentProvider in the query parameter as well? I'm thinking this is the last or one of the last few requests to make in this PR. Thank you @sean-morris for your AMAZING perseverance with this work!!!